frankenphp.c

```diff
@@ -652,106 +654,69 @@ static char *frankenphp_read_cookies(void) {
   return ctx->cookie_data;
 }
 
-static void frankenphp_register_known_variable(const char *key, go_string value,
-                                               zval *track_vars_array) {
-  if (value.data == NULL) {
-    php_register_variable_safe(key, "", 0, track_vars_array);
-    return;
+/* all variables with well defined keys can safely be registered like this */
+void frankenphp_register_trusted_var(zend_string *z_key, char *value,
+                                     int val_len, zval *track_vars_array) {
+  if (value == NULL) {
+    value = "";
   }
-
-  size_t new_val_len;
-  if (sapi_module.input_filter(PARSE_SERVER, key, &value.data, value.len,
-                               &new_val_len)) {
-    php_register_variable_safe(key, value.data, new_val_len, track_vars_array);
+  size_t new_val_len = val_len;
+
+  if (!should_filter_var ||
+      sapi_module.input_filter(PARSE_SERVER, ZSTR_VAL(z_key), &value,
```

Comment:
`$_SERVER` values shouldn't be trusted. For example, `HTTP_*` values are set via headers, some are set via environment variables (which is also user input).

Reply:
Yes, environment variables and headers are registered via `frankenphp_register_variable_safe`. The only variables registered via `frankenphp_register_trusted_var` are those where the keys are pre-defined in the `knownServerKeys` in `cgi.go`.  The keys are therefore hard-coded and the values are also strings that are either pre-defined or sanitized by `net/http` and `filepath.abs`.